packages/graphql/src/mutations/_txHelper.js

```diff
@@ -140,6 +142,437 @@ async function fetchGasPrice() {
   throw new Error(`Gas key of ${GAS_PRICE_KEY} is unavailable`)
 }
 
+/**
+ * Return a globally available web3 instance if available, or die
+ *
+ * @returns {object} web3 instance
+ * @throws {Error} if no global web3 available
+ */
+function globalWeb3() {
+  if (window && window.web3) {
+    return window.web3
+  } else if (global && global.web3) {
+    return global.web3
+  }
+  throw new Error('No available web3')
+}
+
+/**
+ * Get the current going gas price.  If MM is in use, don't use eth_gasPrice,
+ * but fetch from ethgasstation
+ *
+ * @param {object} web3 instance to use (optional)
+ * @returns {number} gas price in wei
+ * @throws {Error} if no web3 given, and no global web3 available
+ */
+async function getGasPrice(w3) {
+  w3 = w3 ? w3 : globalWeb3()
```

Comment:
nit: Not really something that needs changing here, but might be useful in cases where you need to check for states of more than 2 variables: 
```
w3 = w3 || globalWeb3()
// can chain more
w3 = w3 || globalWeb3() || somethingElse()
```